### ISSUE TITLE
Better icon recognition by using OCR on the shortname?

### ISSUE THREAD

**Issue by blitzmann** [RatScanner/RatScanner#72]
Icon recognition is kind of hit or miss at the moment. The little digging I've done in the source shows that it's using trained data to find a decent match, but a lot of the time this gets confused by things like keys (which look very similar) or just plain gets it wrong (I had four magazines side by side in stash, and it evaluated it to a tank battery... (I'll report this as a bug tho as it might be related to the fact that the mages were insured, and thus have a different border color))

I also looked at how you get the legit item from the item info window after clicking the magnifying icon. It looks like you're using OCR on trained data to determine what the text is, and then using that to look up the exact item.

Wouldn't it be possible to apply similar OCR to the icon scan to narrow down what the thing is, by using the shortname? In many cases, that would identify the item correctly from the get go, but for items that have the same shortname but are technically different things (Like `Gpowder`) it can be used to limit down what is considered for icon recognition.

I know nothing about machine learning or how the data is trained to detect characters / icons, limitations of such methods, etc, so I'm interested on hearing about the feasibility of such an improvement, and if it's not feasible, what prevents it from being so? As a developer, I would be interested in digging into it if it's feasible with some guidance, to both improve the software and also my understanding of it

**Comment by Blightbuster** [comment]
I just reread what I wrote down below and see that it can be quite confusing when I talk about icon scanning, template matching, name scanning and shortname scanning. As a quick guidance, when I say icon scanning, I mean the concept of extracting information from a item displayed in the eft inventory system ( which includes the actual icon of the item as well as the shortname). Template matching is what is currently used to detect the item when icon scanning. Shortname scanning ( as you proposed ) would be an alternative for template matching.

When I made the icon scanning, there were two reasons why I did it the way it is currently instead of scanning the shortnames.

1. Applying OCR on the shortnames of icons is defenitly feasible on screens above 90ppi but almost impossible for screens below 70 since the small text smushes together and becomes impossible to read even by humans.

2. Since dynamic icons are cached on the local filesystem, its trivial to know what parts a scanned gun is made of hence allowing for scanning a complete gun instead of just the base part ( which is what you would get with scanning the short name ).

Now this obviously doesnt mean that reading the shortname is of no value ( you clearly brought up a few good arguments for it 😄 ). Infact im working on replacing the whole image processing part of rat scanner with a new library called [Rat Eye](https://github.com/RatScanner/RatEye) which I work on in my off time. For the moment the project structure is pretty much done as well as the implementation of namescanning. Im currently working on implementing the grid detection for the icon scanning. When that is done, it should be straightforward to add the actual template matching for icons ( as an improved version of what is used now ) as well as the proposed shortname scanning of the icons.

I have no idea so far if there will be a performance improvement when combining the shortname scanning and template matching but it will definitly be intresting to perform some benchmarks. But it should definitly be possible to for example use icon scanning ( with the shortname method ) for scanning keys which is not possible by template matching since many keys share the same icon.
